esrally/mechanic/launcher.py

```diff
@@ -376,43 +445,56 @@ def _start_process(self, env, node_name, binary_path):
             raise exceptions.LaunchError("Cannot launch Elasticsearch as root. Please run Rally as a non-root user.")
         os.chdir(binary_path)
         cmd = ["bin/elasticsearch"]
-        return _start(subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, stdin=subprocess.DEVNULL, env=env), node_name)
+        if self.daemon:
+            cmd.extend(["-d", "-p", "pid"])
+            proc = subprocess.Popen(cmd,
+                                    close_fds=True)
+            ret = proc.wait()
+            if ret != 0:
+                msg = "Daemon startup failed with exit code[{}]".format(ret)
+                logging.error(msg)
+                raise exceptions.LaunchError(msg)
+
+            es_log_file = _get_es_log_filename(binary_path)
```

Comment:
We already have a check in place that waits for the cluster to be ready (in `ClusterLauncher`) so I think we can consider a node to be "up" when we detect that the PID file has been written and remove the log file parsing entirely.